ga4gh/frontend.py

```diff
@@ -299,6 +349,56 @@ def assertCorrectVersion(version):
         raise exceptions.VersionNotSupportedException()
 
 
+def startLogin():
+    """
+    If we are not logged in, this generates the redirect URL to the OIDC
+    provider and returns the redirect response
+    :return: A redirect response to the OIDC provider
+    """
+    flask.session["state"] = oic.oauth2.rndstr(SECRET_KEY_LENGTH)
+    flask.session["nonce"] = oic.oauth2.rndstr(SECRET_KEY_LENGTH)
+    args = {
+        "client_id": app.oidcClient.client_id,
+        "response_type": "code",
+        "scope": ["openid", "profile"],
+        "nonce": flask.session["nonce"],
+        "redirect_uri": app.oidcClient.redirect_uris[0],
+        "state": flask.session["state"]
+    }
+
+    result = app.oidcClient.do_authorization_request(
+        request_args=args, state=flask.session["state"])
+    return flask.redirect(result.url)
+
+
+@app.before_request
+def checkAuthentication():
+    """
+    The request will have a parameter 'key' if it came from the command line
+    client, or have a session key of 'key' if it's the browser.
+    If the token is not found, start the login process.
+
+    If there is no oidcClient, we are running naked and we don't check.
+    If we're being redirected to the oidcCallback we don't check.
+
+    :returns nothing if all is ok (and the request handler continues as usual).
```

Comment:
@dcolligan is right. Of course a function will return something, and it will be None. A day you learn something is a good day! Comment fixed.